### PR TITLE
Teaches updateChecker about dev builds

### LINKED ISSUE #3494: updateChecker does not take into account dev builds with `v0.2.0-sha-checksum`
https://github.com/ember-cli/ember-cli/blob/master/lib/models/update-checker.js#L60

cc @twokul

### PR DESCRIPTION
**Pull request by twokul**
fixes #3494

### CODE REVIEW

File: tests/unit/models/update-checker-test.js [hunk 1]
```diff
@@ -57,6 +81,7 @@ describe('Update Checker', function() {
 
         return function() {
           updateChecker.versionConfig = versionConfig;
+          debugger
```
Comment (stefanpenner) [inline tests/unit/models/update-checker-test.js]: Forgotten debugger

**Comment by stefanpenner** [comment]
Awesome

File: lib/models/update-checker.js [hunk 2]
```diff
@@ -56,8 +56,16 @@ UpdateChecker.prototype.doCheck = function() {
   }.bind(this)).catch(function() {
     return this.checkNPM();
   }.bind(this)).then(function(version) {
+    // match postfix SHA in dev version
+    var isDevBuild   = !!this.localVersion.match(/\b[0-9a-f]{5,40}\b/);
```
Comment (rwjblue) [inline lib/models/update-checker.js]: I think I'd rather make [lib/utilities/ember-cli-version.js](https://github.com/ember-cli/ember-cli/blob/master/lib/utilities/ember-cli-version.js) be the only thing that has to be aware of this.  Can we make that script export another function (like `isDevelopment()`) then just consume it here?  

I'm mostly concerned with having the concept of what exactly goes into a dev build version number  be encapsulated in a single place.

Comment (twokul) [inline lib/models/update-checker.js]: @rwjblue makes sense to me. I'll make a change on the flught

**Comment by twokul** [comment]
@stefanpenner @rwjblue I think this is ready for another look

**Comment by rwjblue** [comment]
Thanks!

**Pull request merged**